go/vrf/ecvrf_p256_sha256_tai.go

```diff
@@ -80,3 +80,68 @@ func (a p256SHA256TAIAux) ArbitraryStringToPoint(h []byte) (Px, Py *big.Int, err
 	}
 	return a.StringToPoint(append([]byte{0x02}, h...))
 }
+
+var zero big.Int
+var one = big.NewInt(1)
+
+// HashToCurve implements the HashToCurveTryAndIncrement algorithm from section 5.4.1.1.
+//
+// The running time of this algorithm depends on alpha. For the ciphersuites
+// specified in Section 5.5, this algorithm is expected to find a valid curve
+// point after approximately two attempts (i.e., when ctr=1) on average.
+//
+// However, because the running time of algorithm depends on alpha, this
+// algorithm SHOULD be avoided in applications where it is important that the
+// VRF input alpha remain secret.
+//
+// Inputs:
+// - `suite` - a single octet specifying ECVRF ciphersuite.
+// - `pub`   - public key, an EC point
+// - `alpha` - value to be hashed, an octet string
+// Output:
+// - `H` - hashed value, a finite EC point in G
+// - `ctr` - integer, number of suite byte, attempts to find a valid curve point
+func (a p256SHA256TAIAux) HashToCurve(pub *PublicKey, alpha []byte) (Hx, Hy *big.Int) {
+	Hx, Hy, _ = a.hashToCurve(pub, alpha)
+	return
+}
+
+func (a p256SHA256TAIAux) hashToCurve(pub *PublicKey, alpha []byte) (Hx, Hy *big.Int, ctr byte) {
+	// 1.  ctr = 0
+	ctr = 0
+	// 2.  PK_string = point_to_string(pub)
+	pkStr := a.PointToString(pub.X, pub.Y)
+
+	// 3.  one_string = 0x01 = int_to_string(1, 1), a single octet with value 1
+	oneStr := []byte{0x01}
+
+	// 4.  H = "INVALID"
+	h := a.params.hash.New()
+
+	// 5.  While H is "INVALID" or H is EC point at infinity:
+	var err error
+	for Hx == nil || err != nil || (zero.Cmp(Hx) == 0 && zero.Cmp(Hy) == 0) {
+		// A.  ctr_string = int_to_string(ctr, 1)
+		ctrString := []byte{ctr}
```

Comment:
ctr could silently overflow (although this is extremely unlikely to happen in this specific case), so it would be good to add a bounds check.

Reply:
Good idea.